test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -4202,6 +4208,361 @@ defmodule Trento.SapSystemTest do
     end
   end
 
+  describe "delta deregistration" do
+    test "newly registered instances should be marked as present" do
+      sap_system_id = Faker.UUID.v4()
+      sid = fake_sid()
+      ensa_version = EnsaVersion.ensa1()
+      host_id = Faker.UUID.v4()
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(:application_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          features: "MESSAGESERVER",
+          host_id: host_id
+        ),
+        build(
```

Comment:
This is not correct. We shouldn't have a `sap_system_registered` event unless we have an application server as well